src/freenas/usr/local/lib/middlewared_truenas/plugins/failover_/detect_enclosure_freebsd.py

```diff
@@ -0,0 +1,128 @@
+import subprocess as sb
+import re
+import os
+
+from middlewared.service import private, Service
+from ..failover import HA_HARDWARE
+
+
+class EnclosureDetectionService(Service):
+
+    class Config:
+        namespace = 'failover.enclosure'
+
+    HARDWARE = 'MANUAL'
+    NODE = None
+
+    @private
+    def detect(self):
+
+        # First detect if this is a BHYVE instance
+        manufacturer = sb.run(
+            ['dmidecode -s system-product-name'],
+            stdout=sb.PIPE,
+            stderr=sb.PIPE,
+            shell=True
+        )
+
+        if manufacturer.stdout:
+            manufacturer = manufacturer.stdout.decode().strip()
+
+            if manufacturer == 'BHYVE':
+                self.HARDWARE = 'BHYVE'
+
+            devlist = sb.run(
+                ['/sbin/camcontrol devlist'],
+                stdout=sb.PIPE,
+                stderr=sb.PIPE,
+                shell=True
+            )
+
+            if devlist.stdout:
+                devlist = devlist.stdout.decode()
```

Comment:
If we always decode, better use encoding='utf8', and errors='ignore' in subprocess run